src/NumericalAlgorithms/Spectral/SwshDerivatives.hpp

```diff
@@ -305,5 +305,168 @@ using SwshDerivatives = detail::SwshDerivativesImpl<
     DerivativeTagList,
     typename detail::unique_derived_from_list<DerivativeTagList>::type,
     Representation>;
+
+/*!
+ * \ingroup SpectralGroup
+ * \brief Produces a `SpinWeighted<ComplexModalVector, Spin>` of the appropriate
+ * size to be used as a modal buffer for `Spectral::Swsh::SwshDerivatives` or
+ * `Spectral::Swsh::swsh_derivatives`.
+ *
+ * \details The `Spectral::Swsh::swsh_derivatives` and
+ * `Spectral::Swsh::SwshDerivatives` interfaces require that calling code
+ * provides a buffer for the intermediate transform results, to ensure that
+ * callers are aware of the allocations and can suitably reuse buffers if
+ * possible. This utility eases the creation of those buffers.
+ */
+template <int Spin>
+auto swsh_buffer(const size_t l_max,
+                 const size_t number_of_radial_points) noexcept {
+  return SpinWeighted<ComplexModalVector, Spin>{
+      size_of_libsharp_coefficient_vector(l_max) * number_of_radial_points};
+}
+
+namespace detail {
+// template 'implementation' for the `swsh_derivatives` function below which
+// evaluates an arbitrary number of derivatives, and places them in the set of
+// nodal containers passed by pointer.
+template <ComplexRepresentation Representation, typename... DerivativeKinds,
+          typename... ArgumentTypes, size_t... Is>
+void swsh_derivatives_impl(
+    const size_t l_max, const size_t number_of_radial_points,
+    std::index_sequence<Is...> /*meta*/,
```

Comment:
we normally put the `meta` arguments at the end of the argument list. Is that not possible?